core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1945,11 +1944,13 @@ private SqlCharStringLiteral createDateTimeFormatSqlCharLiteral(String format) {
    */
   private void unparseRandomfunction(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
     SqlCall randCall = RAND.createCall(SqlParserPos.ZERO);
-    SqlCall upperLimitCall = PLUS.createCall(SqlParserPos.ZERO, MINUS.createCall
-            (SqlParserPos.ZERO, call.operand(1), call.operand(0)), call.operand(0));
+    SqlCall upperLimitCall = MINUS.createCall(SqlParserPos.ZERO, call.operand(1), call.operand(0));
     SqlCall numberGenerator = MULTIPLY.createCall(SqlParserPos.ZERO, randCall, upperLimitCall);
-    SqlCall floorDoubleValue = FLOOR.createCall(SqlParserPos.ZERO, numberGenerator);
-    SqlCall plusNode = PLUS.createCall(SqlParserPos.ZERO, floorDoubleValue, call.operand(0));
+    SqlNode integerNode = getCastSpec(
+        new BasicSqlType(RelDataTypeSystem.DEFAULT,
+            SqlTypeName.INTEGER));
+    SqlCall castVlue = CAST.createCall(SqlParserPos.ZERO, numberGenerator, integerNode);
+    SqlCall plusNode = PLUS.createCall(SqlParserPos.ZERO, castVlue, call.operand(0));
```

Comment:
we should handle this in BQConverterRexShuttle

Reply:
Made changes as suggested